OPHD/UI/Core/ToolTip.cpp

```diff
@@ -0,0 +1,80 @@
+#include "ToolTip.h"
+
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/EventHandler.h>
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+
+
+using namespace NAS2D;
+
+
+ToolTip::ToolTip():
+	mFont{ fontCache.load(constants::FONT_PRIMARY, constants::FONT_PRIMARY_NORMAL) }
+{
+	Utility<EventHandler>::get().mouseMotion().connect(this, &ToolTip::onMouseMove);
+}
+
+
+ToolTip::~ToolTip()
+{
+	Utility<EventHandler>::get().mouseMotion().disconnect(this, &ToolTip::onMouseMove);
+}
+
+
+void ToolTip::add(Control& c, const std::string& str)
+{
+	for (auto& item : mControls)
+	{
+		if (item.first == &c)
+		{
+			item.second = str;
+			return;
+		}
+	}
+
+	mControls.push_back(std::make_pair(&c, str));
+}

```

Comment:
You could potentially use a `std::map` with the `Control&` (or `Control*`) as the key. That would eliminate the need to scan the list for updates.

Reply:
True, though I don't see too many cases of duplicate insertion. This just seemed like the lighter-weight way to go versus using a binary tree.